modules/bitcoind.nix

```diff
@@ -407,7 +413,7 @@ in {
         Restart = "on-failure";
         UMask = mkIf cfg.dataDirReadableByGroup "0027";
         ReadWritePaths = cfg.dataDir;
-      } // nbLib.allowedIPAddresses cfg.enforceTor
+      } // nbLib.allowedIPAddresses (cfg.tor.proxy && cfg.tor.enforce)
```

Comment:
This seems to contradict the description of tor.enforc which would suggest that setting tor.enforce alone is enough to enforce tor. Also should we set the default value of `cfg.tor.proxy` to cfg.tor.enforce to make sure that outbound bitcoind options (for example) happen if Tor is enforced?

Instead of having 2 options `enforce` and `proxy` we could alternatively have a single option `tor` with the values "enforce", "out-only", "false"?

Reply:
> This seems to contradict the description of tor.enforc which would suggest that setting tor.enforce alone is enough to enforce tor.

Yes, this is an important point. Fixed in `fixup! tor: don't disable tor enforcement when tor.proxy is disabled`.

> Also should we set the default value of cfg.tor.proxy to cfg.tor.enforce to make sure that outbound bitcoind options (for example) happen if Tor is enforced?

I think it's simpler and clearer if both options are kept orthogonal.

> we could alternatively have a single option tor with the values "enforce", "out-only", "false"?

This is complicated by the fact that some services like `electrs` or `btcpayserver` don't support Tor proxying.
Also, setting `bitcoind.tor = false` looks a bit as this would also disable the hidden service.

Reply:
> I think it's simpler and clearer if both options are kept orthogonal.

You're right.